.travis.yml

```diff
@@ -5,15 +5,26 @@ addons:
     organization: "alfasoftware"
     token:
       secure: "Rk0cQ9JL2/DB8xrKjaU/m9HEvZllokUZ8c7Q+6g3DHwOsMvZi1OoCdUtDXf9Zl0xtdFPMvoWsGPeZ6+sZK9evHscvRH83hVMTrTabWK4+QL660QeMXboBprwuBIpAI0XBCfNcslsT3zPd/e4v35FHPdvRr1yOQlibrjEgs1MZo6/LQ7QgRNJAHsskiqOOJS6LLDHceYJEW7wd/y7R0BvXoxch6xQhDOFvowoc22/cEyUVCSCMMas74hD//veSXdHt0CNOY+H+/PAzAYsbM0rkJnrysqO8UgLZwpnP/Mjsoi0InoisvriSl9MYgQwMAfcrtVxxR94vt4Pu62LXsnt6ufGfwFBrdTh1Rna05fzTAmm9bwywIR2UsMwNeM4iDoz+Rc2ImnwMKRipn5lib5kCOlwqjTv6NVi0oV5XEzTfU51HIrQMmmsq60YwMZgcYwckHoVVOojv29dAjLe0Z/LbfUH8MHprt8/MdWA8ir2egxHHW3dXVY11MgDKzQ2RyYb1uPNJgkVfZDm7uHDcj13dFR61Gy1mTdrtpnWx1ojU2IJQ4oKuGb04RbqQ5lCq7dgsHlaeaAJFMlcn1I6IOk8t6QkYy/3vhSXgkZpQAO3K5Zp4aaGZD90sYdQO3JQNCe2S8mhgEGOqsAdxytFygCaMxOKG5NV9qxVilkzfGPvmqM="
-script:
-  - SONAR_SKIP=$(if [ "$TRAVIS_PULL_REQUEST" = "false" -a $TRAVIS_REPO_SLUG = 'alfasoftware/morf' ]; then echo "false"; else echo "true"; fi)
-  - if [ $SONAR_SKIP = "false" ]; then git fetch --unshallow --quiet; fi
-  - mvn $GOALS -Dsonar.branch=$TRAVIS_BRANCH -Dsonar.skip=$SONAR_SKIP
+jobs:
+  include:
+    - if: branch = master AND repo = alfasoftware/morf AND NOT type = pull_request
+      name: "Test with coverage"
+      script:
+        - mvn clean org.jacoco:jacoco-maven-plugin:prepare-agent test sonar:sonar -Pcoverage-per-test -Dsonar.branch.name=master -B -U
+    - if: branch = master AND repo = alfasoftware/morf AND NOT type = pull_request
+      name: "Deploy"
+      script:
+        - echo $GPG_SECRET_KEYS | base64 --decode | $GPG_EXECUTABLE --import
+        - echo $GPG_OWNERTRUST | base64 --decode | $GPG_EXECUTABLE --import-ownertrust
+        - mvn clean deploy --settings .maven-settings.xml -B -U -Prelease

```

Comment:
Can we not create a profile which does both the testing with coverage _and_ release and avoid the two separate `mvn` calls?

Reply:
We could, thought it would be better to keep them separate to reduce the build time (as they run in parallel) and the build output so that we don't exceed the allowed limit on Travis. 

The test + coverage job now runs mvn test (not package), so it doesn't do the unnecessary packaging work it used to do before, where as the deploy task focuses on essentials for publishing. Typically people add -DskipTests to deploy commands, which we could as well, but I felt that's probably not a good idea (but we have the ability to do that if we need to).

I'm not attached to either approaches really :) @badgerwithagun, @wnicholson, thoughts?

Reply:
Meh, this is fine for now. We can always change our minds!